pebblo/app/service/local_ui/loader_apps.py

```diff
@@ -168,28 +183,32 @@ def _update_loader_datasource(
         """
         This function updates loader datasource details and count
         """
-
         _, data_sources = self.db.query(
             AiDataSourceTable, {"loadId": app_data.get("id")}
         )
         for data_source in data_sources:
-            ds_data = data_source.data
-            ds_obj = {
-                "appName": ds_data["appName"],
-                "name": ds_data["loader"],
-                "sourcePath": ds_data["sourcePath"],
-                "sourceType": ds_data["sourceType"],
-                "findingsEntities": entity_count,
-                "findingsTopics": topic_count,
-                "totalSnippetCount": total_snippet_count,
-                "displayedSnippetCount": min(
-                    ReportConstants.SNIPPET_LIMIT.value, total_snippet_count
-                ),
-            }
-            self.loader_data_source_list.append(ds_obj)
+            try:
+                ds_data = data_source.data
+                ds_obj = {
+                    "appName": ds_data["appName"],
+                    "name": ds_data["loader"],
+                    "sourcePath": ds_data["sourcePath"],
+                    "sourceType": ds_data["sourceType"],
+                    "findingsEntities": entity_count,
+                    "findingsTopics": topic_count,
+                    "totalSnippetCount": total_snippet_count,
+                    "displayedSnippetCount": min(
+                        ReportConstants.SNIPPET_LIMIT.value, total_snippet_count
+                    ),
+                }
+                self.loader_data_source_list.append(ds_obj)
+            except Exception as err:
+                logger.error(
+                    f"Failed in getting data source details of {data_source.data}, Error: {err}"
+                )
 
-        # Data Source Count
-        self.loader_data_source = len(self.loader_data_source_list)
+            # Documents with findings Count
```

Comment:
Don't we need this

`self.loader_data_source = len(self.loader_data_source_list)`

Reply:
Fixed.